src/inmanta/data/model.py

```diff
@@ -898,6 +898,19 @@ def has_source(self) -> bool:
 LEGACY_PIP_DEFAULT = PipConfig(use_system_config=True)
 
 
+class SchedulerStatusReport(BaseModel):
+    """
+    Status report for the scheduler self-check
+
+    :param resource_state: In-memory representation of the resources in the scheduler
+    :param discrepancies: Discrepancies between the in-memory representation of the resources
+        and their state in the database.
+    """
+
+    resource_state: dict[ResourceIdStr, typing.Any]
+    discrepancies: dict[ResourceIdStr, typing.Any]
```

Comment:
It would be nicer if the second arguments are typed, not crucial

Reply:
I don't get how to type these, there's probably something obvious I'm missing. 
e.g. for `resource_state` 
Should we import ResourceState from inmanta.deploy.state and type it as such here ? I tried but I was running into circular imports but I can have another look if that's the way to go.

`    resource_state: dict[ResourceIdStr, ResourceState]`